server/game/game.js

```diff
@@ -1225,6 +1229,18 @@ class Game extends EventEmitter {
         return 'game';
     }
 
+    pauseClock() {
+        this.clockPaused = !this.clockPaused;
+        if(this.useChessClocks) {
+            let players = this.getPlayers();
+            for(let player of players) {
+                player.togglePauseChessClock();
+            }
+        }
+
+        //TODO pause time limit clock, not only chess clock
```

Comment:
I'm wondering in general if there's a way for the clock mechanism to be generalized. e.g. use the same interface for both time limit clock and chess clock, so that the game logic doesn't need to care so much about the details of which clock is being used.

Reply:
The Keyteki project is doing that (as they have even more types of clocks) but it does make the code a bit more complicated as to what part of code is doing work for which type of clock. But using at least the same interface would be a good idea. I will open up a follow-up issue for that.